server/src/main/java/org/opensearch/rest/BytesRestResponse.java

```diff
@@ -107,22 +109,30 @@ public BytesRestResponse(RestChannel channel, Exception e) throws IOException {
 
     public BytesRestResponse(RestChannel channel, RestStatus status, Exception e) throws IOException {
         ToXContent.Params params = paramsFromRequest(channel.request());
+        boolean doubleSlashPresent = checkForDoubleSlash(channel.request().rawPath());
```

Comment:
The double slash check strikes me as a special case when it might be better to handle this more generally. It looks to me that the system correctly parsed the request and identified that the request specified the empty string as the index name. Where it went wrong is that it surfaced an "index out of bounds" failure as opposed to a more sensible parameter validation error (i.e. "No index name specified"). I'm concerned that this double slash check will have unexpected behavior, such as if there are cases where a double slash somewhere in the URL is either acceptable or at least innocuous and this change would be a backwards incompatible change in behavior. Can you fix this bug by doing more strict checks of the index name itself as opposed to looking for double slashes in the URL path?

Reply:
It's correct that checking double slash is not a good solution, because currently the REST APIs have got the leniency to allow double slash in URL.
However, after taking a deeper look at the error message. I don't think the system identified that the request contains an empty "index" name.
1. There is no `PUT <index>/_cluster/health` API, and as the author of the issue said, adding a leading `/` to any valid `PUT /_x/y` APIs will result that error message.
2. The error message only occurs using `curl` command, but will not occur when using the in Dashboards Console (https://www.elastic.co/guide/en/kibana/7.10/console-kibana.html)
3. Looks like the `string_index_out_of_bounds_exception` comes from Java (https://docs.oracle.com/javase/8/docs/api/java/lang/StringIndexOutOfBoundsException.html), and are parsed by OpenSearch to add the underscores (https://github.com/opensearch-project/OpenSearch/blob/1.1.0/server/src/main/java/org/opensearch/OpenSearchException.java#L688)

So I think we should find a better solution for this issue. 😁

Reply:
I agree with the checking double slash as not being a good solution. Also, as the system does not identify the empty index name as well as the author opening the issue related to empty index or double slash, checking for it's value being null is the change, I have made. Correct me if I am wrong here.